src/error.rs

```diff
@@ -21,6 +30,43 @@ pub type DaftResult<T> = std::result::Result<T, DaftError>;
 impl Display for DaftError {
     // `f` is a buffer, and this method must write the formatted string into it
     fn fmt(&self, f: &mut Formatter) -> Result {
-        write!(f, "{self:?}")
+        // Override this method if the error propagated should have custom display for better user-level error messages
+        match self {
+            Self::ExprResolveTypeError {
+                expectation,
+                op_display_name,
+                binary_op_display,
+                fields_to_expr,
+            } => {
+                let simple_field_display = match (binary_op_display, &fields_to_expr[..]) {
+                    (Some(op), [(left_field, _), (right_field, _)]) => {
+                        format!("`{}` {op} `{}`", left_field.name, right_field.name)
+                    }
+                    _ => {
+                        let args: Vec<String> = fields_to_expr
+                            .iter()
+                            .map(|(field, _)| format!("`{}`", field.name))
+                            .collect();
+                        format!("{op_display_name}({})", args.join(", "))
```

Comment:
this should also be implemented for other ops? Why are we reimplementing it?

Reply:
See above comment:

> I want to print just `{field_name} {op} {field_name}` but the binary_op Display would print `{left_expr} {op} {right_expr}`

In this case, the other expr Displays would print something like `sum({child_expr})`, but I just want `sum(`field_name`)`.

Reply:
is this completely needed? I really don't like how we have Display logic for Expr somewhere else in the code base.

Reply:
So the difference is a user-facing error like:

```
TypeError: - expects left and right arguments to both be numeric, but failed type resolution: [col("x") - 1].alias("foo") - col("y")
```

vs

```
TypeError: - expects left and right arguments to both be numeric, but failed type resolution: `foo` - `y`
where:
  `foo` resolves to UInt8: [col("x") - 1].alias("foo")
  `y` resolves to Utf8:  col("y")
```


1. In the first message, it isn't entirely clear which `-` is causing the issue
2. The second message has the resolved types of the arguments as well, which is nice visualization

WDYT?

Reply:
the `where:` part can happen on both of them without the custom repr right?
its really just: `[col("x") - 1].alias("foo") - col("y")` vs  `foo` - `y`

Reply:
so it would be more like 
```
    ExprResolveTypeError {
        expectation: String,
        expression: Expr,
        fields_to_expr: Vec<(Field, Arc<Expr>)>,
    },
```
Which is much cleaner imo, and doesn't require duplicated repr logic

Reply:
Ok, pushed changes